### PR TITLE
Validate and document minimum  max pool size for JDBC PING cache stack

### LINKED ISSUE #46673: Raise minimum maximum number of `poolMaxSize` as value 3 shown in Keycloak example leads to acquisition timeout after switch to JDBC Ping
### Before reporting an issue

- [x] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

operator

### Describe the bug

When you follow documentation example (which should not be taken literally) in https://www.keycloak.org/operator/advanced-configuration#_server_configuration_details you have db config like:

```yaml
spec:
  db:
    poolInitialSize: 1
    poolMinSize: 2
    poolMaxSize: 3
```

However, after `cache-stack` default switch from K8 to JDBC PING in #40201 (cc @ryanemerson ), having 3 connection may not be enough on startup as there seem to be nested transactions in `runJobInTransaction`, for example:

```
JGroupsConfigurator.configureDiscovery (line 251)                                                 
    ↓ starts TRANSACTION 1 via KeycloakModelUtils.runJobInTransactionWithResult                     
      ↓ calls prepareJGroupsAddress (line 270)                                                      
        ↓ calls getNextSequence (line 297)                                                          
          ↓ starts TRANSACTION 2 (NESTED!) via KeycloakModelUtils.runJobInTransactionWithResult
            ↓ tries to access database via JpaServerConfigStorageProvider.loadOrCreate
              ↓ Hibernate tries to acquire connection...  --> leads to timeout
```

And thus, application startup fails. 

### Version

26.4.7+ (that is what I verified, probably 26.3.x+, give it a try)

### Regression

- [x] The issue is a regression

### Expected behavior

At the very least, we need to adjust the documentation example. But raising minimal value of the `poolMaxSize` would be highly desirable and we need to test that our new minimum works as expected.

### Actual behavior

Application fails to start. I can see following stacktrace:

```
2026-02-25 18:57:17,634 INFO  [org.hibernate.event.internal.DefaultLoadEventListener] (main) HHH000327: Error performing load command: org.hibernate.exception.GenericJDBCException: Unable to acquire JDBC Connection [Sorry, acquisition timeout!] [n/a]
	at org.hibernate.exception.internal.StandardSQLExceptionConverter.convert(StandardSQLExceptionConverter.java:39)
	at org.hibernate.engine.jdbc.spi.SqlExceptionHelper.convert(SqlExceptionHelper.java:115)
	at org.hibernate.engine.jdbc.spi.SqlExceptionHelper.convert(SqlExceptionHelper.java:101)
	at org.hibernate.resource.jdbc.internal.LogicalConnectionManagedImpl.acquire(LogicalConnectionManagedImpl.java:190)
	at org.hibernate.resource.jdbc.internal.LogicalConnectionManagedImpl.acquireConnectionIfNeeded(LogicalConnectionManagedImpl.java:87)
	at org.hibernate.resource.jdbc.internal.LogicalConnectionManagedImpl.getPhysicalConnection(LogicalConnectionManagedImpl.java:126)
	at org.hibernate.engine.jdbc.internal.StatementPreparerImpl.connection(StatementPreparerImpl.java:57)
	at org.hibernate.engine.jdbc.internal.StatementPreparerImpl$4.doPrepare(StatementPreparerImpl.java:158)
	at org.hibernate.engine.jdbc.internal.StatementPreparerImpl$StatementPreparationTemplate.prepareStatement(StatementPreparerImpl.java:187)
	at org.hibernate.engine.jdbc.internal.StatementPreparerImpl.prepareQueryStatement(StatementPreparerImpl.java:160)
	at org.hibernate.sql.exec.internal.StandardStatementCreator.createStatement(StandardStatementCreator.java:47)
	at org.hibernate.sql.results.jdbc.internal.DeferredResultSetAccess.executeQuery(DeferredResultSetAccess.java:271)
	at org.hibernate.sql.results.jdbc.internal.DeferredResultSetAccess.getResultSet(DeferredResultSetAccess.java:200)
	at org.hibernate.sql.results.jdbc.internal.JdbcValuesResultSetImpl.<init>(JdbcValuesResultSetImpl.java:72)
	at org.hibernate.sql.exec.internal.JdbcSelectExecutorStandardImpl.resolveJdbcValues(JdbcSelectExecutorStandardImpl.java:372)
	at org.hibernate.sql.exec.internal.JdbcSelectExecutorStandardImpl.resolveJdbcValuesSource(JdbcSelectExecutorStandardImpl.java:332)
	at org.hibernate.sql.exec.internal.JdbcSelectExecutorStandardImpl.doExecuteQuery(JdbcSelectExecutorStandardImpl.java:135)
	at org.hibernate.sql.exec.internal.JdbcSelectExecutorStandardImpl.executeQuery(JdbcSelectExecutorStandardImpl.java:100)
	at org.hibernate.sql.exec.spi.JdbcSelectExecutor.executeQuery(JdbcSelectExecutor.java:64)
	at org.hibernate.sql.exec.spi.JdbcSelectExecutor.list(JdbcSelectExecutor.java:138)
	at org.hibernate.loader.ast.internal.SingleIdLoadPlan.load(SingleIdLoadPlan.java:143)
	at org.hibernate.loader.ast.internal.SingleIdEntityLoaderStandardImpl.load(SingleIdEntityLoaderStandardImpl.java:77)
	at org.hibernate.loader.ast.internal.AbstractEntityBatchLoader.load(AbstractEntityBatchLoader.java:92)
	at org.hibernate.loader.ast.internal.AbstractEntityBatchLoader.load(AbstractEntityBatchLoader.java:54)
	at org.hibernate.loader.ast.internal.EntityBatchLoaderArrayParam.load(EntityBatchLoaderArrayParam.java:166)
	at org.hibernate.persister.entity.AbstractEntityPersister.doLoad(AbstractEntityPersister.java:3532)
	at org.hibernate.persister.entity.AbstractEntityPersister.load(AbstractEntityPersister.java:3521)
	at org.hibernate.event.internal.DefaultLoadEventListener.loadFromDatasource(DefaultLoadEventListener.java:596)
	at org.hibernate.event.internal.DefaultLoadEventListener.loadFromCacheOrDatasource(DefaultLoadEventListener.java:570)
	at org.hibernate.event.internal.DefaultLoadEventListener.load(DefaultLoadEventListener.java:554)
	at org.hibernate.event.internal.DefaultLoadEventListener.doLoad(DefaultLoadEventListener.java:538)
	at org.hibernate.event.internal.DefaultLoadEventListener.load(DefaultLoadEventListener.java:204)
	at org.hibernate.event.internal.DefaultLoadEventListener.lockAndLoad(DefaultLoadEventListener.java:489)
	at org.hibernate.event.internal.DefaultLoadEventListener.doOnLoad(DefaultLoadEventListener.java:110)
	at org.hibernate.event.internal.DefaultLoadEventListener.onLoad(DefaultLoadEventListener.java:68)
	at org.hibernate.event.service.internal.EventListenerGroupImpl.fireEventOnEachListener(EventListenerGroupImpl.java:151)
	at org.hibernate.internal.SessionImpl.fireLoadNoChecks(SessionImpl.java:1291)
	at org.hibernate.internal.SessionImpl.fireLoad(SessionImpl.java:1278)
	at org.hibernate.internal.SessionImpl.load(SessionImpl.java:1251)
	at org.hibernate.loader.internal.IdentifierLoadAccessImpl.doLoad(IdentifierLoadAccessImpl.java:181)
	at org.hibernate.loader.internal.IdentifierLoadAccessImpl.lambda$load$1(IdentifierLoadAccessImpl.java:167)
	at org.hibernate.loader.internal.IdentifierLoadAccessImpl.perform(IdentifierLoadAccessImpl.java:134)
	at org.hibernate.loader.internal.IdentifierLoadAccessImpl.load(IdentifierLoadAccessImpl.java:167)
	at org.hibernate.internal.SessionImpl.find(SessionImpl.java:2407)
	at org.hibernate.internal.SessionImpl.find(SessionImpl.java:2397)
	at org.hibernate.internal.SessionImpl.find(SessionImpl.java:2386)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at org.keycloak.connections.jpa.support.EntityManagerProxy.invoke(EntityManagerProxy.java:100)
	at jdk.proxy2/jdk.proxy2.$Proxy57.find(Unknown Source)
	at org.keycloak.storage.configuration.jpa.JpaServerConfigStorageProvider.getEntity(JpaServerConfigStorageProvider.java:105)
	at org.keycloak.storage.configuration.jpa.JpaServerConfigStorageProvider.loadOrCreate(JpaServerConfigStorageProvider.java:71)
	at org.keycloak.spi.infinispan.impl.embedded.JGroupsConfigurator.lambda$getNextSequence$6(JGroupsConfigurator.java:299)
	at org.keycloak.models.utils.KeycloakModelUtils.runJobInTransactionWithResult(KeycloakModelUtils.java:461)
	at org.keycloak.models.utils.KeycloakModelUtils.runJobInTransactionWithResult(KeycloakModelUtils.java:439)
	at org.keycloak.spi.infinispan.impl.embedded.JGroupsConfigurator.getNextSequence(JGroupsConfigurator.java:297)
	at org.keycloak.spi.infinispan.impl.embedded.JGroupsConfigurator.prepareJGroupsAddress(JGroupsConfigurator.java:270)
	at org.keycloak.spi.infinispan.impl.embedded.JGroupsConfigurator.lambda$configureDiscovery$2(JGroupsConfigurator.java:251)
	at org.keycloak.models.utils.KeycloakModelUtils.runJobInTransactionWithResult(KeycloakModelUtils.java:461)
	at org.keycloak.models.utils.KeycloakModelUtils.runJobInTransactionWithResult(KeycloakModelUtils.java:439)
	at org.keycloak.spi.infinispan.impl.embedded.JGroupsConfigurator.lambda$configureDiscovery$3(JGroupsConfigurator.java:250)
	at org.keycloak.common.util.Retry.call(Retry.java:149)
	at org.keycloak.spi.infinispan.impl.embedded.JGroupsConfigurator.configureDiscovery(JGroupsConfigurator.java:250)
	at org.keycloak.spi.infinispan.impl.embedded.JGroupsConfigurator.configureJGroups(JGroupsConfigurator.java:149)
	at org.keycloak.spi.infinispan.impl.embedded.DefaultCacheEmbeddedConfigProviderFactory.lambda$singleSiteConfiguration$2(DefaultCacheEmbeddedConfigProviderFactory.java:229)
	at org.keycloak.models.utils.KeycloakModelUtils.lambda$runJobInTransaction$1(KeycloakModelUtils.java:347)
	at org.keycloak.models.utils.KeycloakModelUtils.runJobInTransactionWithResult(KeycloakModelUtils.java:461)
	at org.keycloak.models.utils.KeycloakModelUtils.runJobInTransaction(KeycloakModelUtils.java:346)
	at org.keycloak.models.utils.KeycloakModelUtils.runJobInTransaction(KeycloakModelUtils.java:336)
	at org.keycloak.spi.infinispan.impl.embedded.DefaultCacheEmbeddedConfigProviderFactory.singleSiteConfiguration(DefaultCacheEmbeddedConfigProviderFactory.java:229)
	at org.keycloak.spi.infinispan.impl.embedded.DefaultCacheEmbeddedConfigProviderFactory.configureSingleSiteWithPersistentSessions(DefaultCacheEmbeddedConfigProviderFactory.java:178)
	at org.keycloak.spi.infinispan.impl.embedded.DefaultCacheEmbeddedConfigProviderFactory.createConfiguration(DefaultCacheEmbeddedConfigProviderFactory.java:166)
	at org.keycloak.spi.infinispan.impl.embedded.DefaultCacheEmbeddedConfigProviderFactory.lazyInit(DefaultCacheEmbeddedConfigProviderFactory.java:153)
	at org.keycloak.spi.infinispan.impl.embedded.DefaultCacheEmbeddedConfigProviderFactory.create(DefaultCacheEmbeddedConfigProviderFactory.java:91)
	at org.keycloak.spi.infinispan.impl.embedded.DefaultCacheEmbeddedConfigProviderFactory.create(DefaultCacheEmbeddedConfigProviderFactory.java:69)
	at org.keycloak.services.DefaultKeycloakSession.getOrCreateProvider(DefaultKeycloakSession.java:183)
	at org.keycloak.services.DefaultKeycloakSession.getProvider(DefaultKeycloakSession.java:172)
	at org.keycloak.connections.infinispan.DefaultInfinispanConnectionProviderFactory.createEmbeddedCacheManager(DefaultInfinispanConnectionProviderFactory.java:185)
	at org.keycloak.connections.infinispan.DefaultInfinispanConnectionProviderFactory.lazyInit(DefaultInfinispanConnectionProviderFactory.java:169)
	at org.keycloak.connections.infinispan.DefaultInfinispanConnectionProviderFactory.create(DefaultInfinispanConnectionProviderFactory.java:101)
	at org.keycloak.connections.infinispan.DefaultInfinispanConnectionProviderFactory.create(DefaultInfinispanConnectionProviderFactory.java:87)
	at org.keycloak.services.DefaultKeycloakSession.getOrCreateProvider(DefaultKeycloakSession.java:183)
	at org.keycloak.services.DefaultKeycloakSession.getProvider(DefaultKeycloakSession.java:172)
	at org.keycloak.models.cache.infinispan.InfinispanCacheRealmProviderFactory.lazyInit(InfinispanCacheRealmProviderFactory.java:55)
	at org.keycloak.models.cache.infinispan.InfinispanCacheRealmProviderFactory.create(InfinispanCacheRealmProviderFactory.java:47)
	at org.keycloak.models.cache.infinispan.InfinispanCacheRealmProviderFactory.create(InfinispanCacheRealmProviderFactory.java:37)
	at org.keycloak.services.DefaultKeycloakSession.getOrCreateProvider(DefaultKeycloakSession.java:183)
	at org.keycloak.services.DefaultKeycloakSession.getProvider(DefaultKeycloakSession.java:172)
	at org.keycloak.storage.datastore.DefaultDatastoreProvider.getRealmProvider(DefaultDatastoreProvider.java:156)
	at org.keycloak.storage.datastore.DefaultDatastoreProvider.realms(DefaultDatastoreProvider.java:234)
	at org.keycloak.services.DefaultKeycloakSession.realms(DefaultKeycloakSession.java:257)
	at org.keycloak.migration.migrators.MigrateTo1_2_0.migrate(MigrateTo1_2_0.java:69)
	at org.keycloak.storage.datastore.DefaultMigrationManager.migrate(DefaultMigrationManager.java:165)
	at org.keycloak.migration.MigrationModelManager.migrate(MigrationModelManager.java:33)
	at org.keycloak.quarkus.runtime.storage.database.jpa.QuarkusJpaConnectionProviderFactory.migrateModel(QuarkusJpaConnectionProviderFactory.java:205)
	at org.keycloak.quarkus.runtime.storage.database.jpa.QuarkusJpaConnectionProviderFactory.initSchema(QuarkusJpaConnectionProviderFactory.java:196)
	at org.keycloak.models.utils.KeycloakModelUtils.lambda$runJobInTransaction$1(KeycloakModelUtils.java:347)
	at org.keycloak.models.utils.KeycloakModelUtils.runJobInTransactionWithResult(KeycloakModelUtils.java:461)
	at org.keycloak.models.utils.KeycloakModelUtils.runJobInTransaction(KeycloakModelUtils.java:346)
	at org.keycloak.models.utils.KeycloakModelUtils.runJobInTransaction(KeycloakModelUtils.java:336)
	at org.keycloak.quarkus.runtime.storage.database.jpa.QuarkusJpaConnectionProviderFactory.postInit(QuarkusJpaConnectionProviderFactory.java:129)
	at org.keycloak.services.DefaultKeycloakSessionFactory.initializeProviders(DefaultKeycloakSessionFactory.java:167)
	at org.keycloak.services.DefaultKeycloakSessionFactory.initializeProviders(DefaultKeycloakSessionFactory.java:164)
	at org.keycloak.services.DefaultKeycloakSessionFactory.initializeProviders(DefaultKeycloakSessionFactory.java:164)
	at org.keycloak.services.DefaultKeycloakSessionFactory.initializeProviders(DefaultKeycloakSessionFactory.java:164)
	at org.keycloak.services.DefaultKeycloakSessionFactory.initProviderFactories(DefaultKeycloakSessionFactory.java:141)
	at org.keycloak.services.DefaultKeycloakSessionFactory.initProviderFactories(DefaultKeycloakSessionFactory.java:125)
	at org.keycloak.quarkus.runtime.integration.QuarkusKeycloakSessionFactory.init(QuarkusKeycloakSessionFactory.java:87)
	at org.keycloak.quarkus.runtime.integration.jaxrs.QuarkusKeycloakApplication.createSessionFactory(QuarkusKeycloakApplication.java:60)
	at org.keycloak.services.resources.KeycloakApplication.startup(KeycloakApplication.java:75)
	at org.keycloak.quarkus.runtime.integration.jaxrs.QuarkusKeycloakApplication.onStartupEvent(QuarkusKeycloakApplication.java:50)
	at org.keycloak.quarkus.runtime.integration.jaxrs.QuarkusKeycloakApplication_Observer_onStartupEvent_VUhusbW4xizn9A5dimKJUcfvjHs.notify(Unknown Source)
	at io.quarkus.arc.impl.EventImpl$Notifier.notifyObservers(EventImpl.java:365)
	at io.quarkus.arc.impl.EventImpl$Notifier.notify(EventImpl.java:347)
	at io.quarkus.arc.impl.EventImpl.fire(EventImpl.java:81)
	at io.quarkus.arc.runtime.ArcRecorder.fireLifecycleEvent(ArcRecorder.java:163)
	at io.quarkus.arc.runtime.ArcRecorder.handleLifecycleEvents(ArcRecorder.java:114)
	at io.quarkus.runner.recorded.LifecycleEventsBuildStep$startupEvent1144526294.deploy_0(Unknown Source)
	at io.quarkus.runner.recorded.LifecycleEventsBuildStep$startupEvent1144526294.deploy(Unknown Source)
	at io.quarkus.runner.ApplicationImpl.doStart(Unknown Source)
	at io.quarkus.runtime.Application.start(Application.java:101)
	at io.quarkus.runtime.ApplicationLifecycleManager.run(ApplicationLifecycleManager.java:119)
	at io.quarkus.runtime.Quarkus.run(Quarkus.java:80)
	at org.keycloak.quarkus.runtime.KeycloakMain.start(KeycloakMain.java:139)
	at org.keycloak.quarkus.runtime.cli.Picocli.start(Picocli.java:913)
	at org.keycloak.quarkus.runtime.cli.command.AbstractAutoBuildCommand.runCommand(AbstractAutoBuildCommand.java:134)
	at org.keycloak.quarkus.runtime.cli.command.AbstractCommand.lambda$call$1(AbstractCommand.java:73)
	at java.base/java.util.Optional.orElseGet(Optional.java:364)
	at org.keycloak.quarkus.runtime.cli.command.AbstractCommand.call(AbstractCommand.java:72)
	at org.keycloak.quarkus.runtime.cli.command.AbstractCommand.call(AbstractCommand.java:37)
	at picocli.CommandLine.executeUserObject(CommandLine.java:2031)
	at picocli.CommandLine.access$1500(CommandLine.java:148)
	at picocli.CommandLine$RunLast.executeUserObjectOfLastSubcommandWithSameParent(CommandLine.java:2469)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2461)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2423)
	at picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:2277)
	at picocli.CommandLine$RunLast.execute(CommandLine.java:2425)
	at picocli.CommandLine.execute(CommandLine.java:2174)
	at org.keycloak.quarkus.runtime.cli.Picocli.parseAndRun(Picocli.java:118)
	at org.keycloak.quarkus.runtime.KeycloakMain.main(KeycloakMain.java:109)
	at org.keycloak.quarkus.runtime.KeycloakMain.main(KeycloakMain.java:83)
	at io.quarkus.bootstrap.runner.QuarkusEntryPoint.doRun(QuarkusEntryPoint.java:69)
	at io.quarkus.bootstrap.runner.QuarkusEntryPoint.main(QuarkusEntryPoint.java:37)
Caused by: java.sql.SQLException: Sorry, acquisition timeout!
	at io.agroal.pool.ConnectionPool.waitAvailableHandler(ConnectionPool.java:403)
	at io.agroal.pool.ConnectionPool.handlerFromSharedCache(ConnectionPool.java:364)
	at io.agroal.pool.ConnectionPool.getConnection(ConnectionPool.java:308)
	at io.agroal.pool.DataSource.getConnection(DataSource.java:86)
	at io.quarkus.hibernate.orm.runtime.customized.QuarkusConnectionProvider.getConnection(QuarkusConnectionProvider.java:27)
	at org.hibernate.internal.NonContextualJdbcConnectionAccess.obtainConnection(NonContextualJdbcConnectionAccess.java:44)
	at org.hibernate.resource.jdbc.internal.LogicalConnectionManagedImpl.acquire(LogicalConnectionManagedImpl.java:187)
	... 139 more
```

### How to Reproduce?

I have prepared automated reproducer, you can reuse it for yourself: https://github.com/michalvavrik/keycloak/actions/runs/22411231634

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by michalvavrik**
Currently when migration tasks are run, we see that the app startup fails with DB pool size 3 or less for JDBC_PING. Hence, we started to validate that the minimal max pool size is 4 for JDBC PING.

* Closes: https://github.com/keycloak/keycloak/issues/46673

### CODE REVIEW

File: docs/guides/operator/advanced-configuration.adoc [hunk 1]
```diff
@@ -59,7 +59,8 @@ spec:
     xaEnabled: false
 ----
 
-For a list of options, see the Keycloak CRD. For details on configuring options, see <@links.server id="all-config"/>.
+The values in this example are illustrative and should not be used literally in production.
```
Comment (mabartos) [inline docs/guides/operator/advanced-configuration.adoc]:
```suggestion
WARNING: The values in this example are illustrative and should not be used literally in production.

```
I'd recommend using warning like this.

File: quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/configuration/mappers/DatabasePropertyMappers.java [hunk 2]
```diff
@@ -22,16 +26,19 @@
...
 import static org.keycloak.config.DatabaseOptions.Datasources.getKeyForDatasource;
+import static org.keycloak.quarkus.runtime.configuration.Configuration.getOptionalKcValue;
 import static org.keycloak.quarkus.runtime.configuration.MicroProfileConfigProvider.NS_KEYCLOAK_PREFIX;
 import static org.keycloak.quarkus.runtime.configuration.mappers.DatabasePropertyMappers.Datasources.appendDatasourceMappers;
 import static org.keycloak.quarkus.runtime.configuration.mappers.PropertyMapper.fromOption;
 
 public final class DatabasePropertyMappers implements PropertyMapperGrouping {
     public static final String PG_TARGET_SERVER_TYPE = "quarkus.datasource.jdbc.additional-jdbc-properties.targetServerType";
     private static final Logger log = Logger.getLogger(DatabasePropertyMappers.class);
+    private static final int JDBC_PING_MIN_POOL_MAX_SIZE = 4;
```
Comment (mabartos) [inline quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/configuration/mappers/DatabasePropertyMappers.java:47]: It'd be nice to put a little comment why it's exactly 4 - maybe just mention that it was based on experiments.

**Review by mabartos — COMMENTED** [review]
@michalvavrik Nice work! Just added a few nitpicks. 

@ahus1 @shawkins @vmuzikar WDYT?

File: docs/guides/operator/advanced-configuration.adoc (hunk 1 above)
Comment (ahus1) [reply docs/guides/operator/advanced-configuration.adoc]: I'd say "The values in this example are illustrative and should not be used literally in production." is a cheap disclaimer. How should a user know what are good values if now by looking at our docs?

File: docs/guides/operator/advanced-configuration.adoc [hunk 3]
```diff
@@ -37,7 +37,7 @@ spec:
     schema: schema
     poolInitialSize: 1
     poolMinSize: 2
-    poolMaxSize: 3
+    poolMaxSize: 99
```
Comment (ahus1) [inline docs/guides/operator/advanced-configuration.adoc]: I'd suggest maybe 30 connections in the pool. Given that people would run 3 instances, this leads to 90 connections, while otherwise one would have 297 connections which is usually way to much. 

```suggestion
    poolMaxSize: 30
```

**Review by ahus1 — CHANGES_REQUESTED** [review]
Thank you for the docs PR. One nitpick about the pool size and the warning, otherwise fine to merge.

File: docs/guides/operator/advanced-configuration.adoc (hunk 1 above)
Comment (michalvavrik) [reply docs/guides/operator/advanced-configuration.adoc]: > I'd say "The values in this example are illustrative and should not be used literally in production." is a cheap disclaimer.

It is a cheap disclaimer, but when I complained that I saw it in the docs, @shawkins told me that these values shouldn't be taken literally, so I think we should tell users as well. Right below this is a sentence where they can find docs for properties.

I'll try to rewrite to something that basically says: they are illustrative values and you should pick values based on your use case and respective documentation (like Quarkus / Agroal docs etc.). Let's see.

Comment (michalvavrik) [reply docs/guides/operator/advanced-configuration.adoc]: > How should a user know what are good values if now by looking at our docs?

If the paragraph right after that doesn't help, we would need a list of respective documentation for each configuration value. It would be better to have it in the `<@links.server id="all-config"/>` than duplicate that

File: docs/guides/operator/advanced-configuration.adoc (hunk 3 above)
Comment (michalvavrik) [reply docs/guides/operator/advanced-configuration.adoc]: > I'd suggest maybe 30 connections in the pool. Given that people would run 3 instances, this leads to 90 connections, while otherwise one would have 297 connections which is usually way to much.

makes sense, thanks

File: quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/configuration/mappers/DatabasePropertyMappers.java (hunk 2 above)
Comment (michalvavrik) [reply quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/configuration/mappers/DatabasePropertyMappers.java:47]: > It'd be nice to put a little comment why it's exactly 4 - maybe just mention that it was based on experiments.

on it, thanks

File: docs/guides/operator/advanced-configuration.adoc (hunk 1 above)
Comment (michalvavrik) [reply docs/guides/operator/advanced-configuration.adoc]: Done. Let's check if it better. Thanks

File: quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/configuration/mappers/DatabasePropertyMappers.java (hunk 2 above)
Comment (michalvavrik) [reply quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/configuration/mappers/DatabasePropertyMappers.java:47]: done

File: docs/guides/operator/advanced-configuration.adoc (hunk 3 above)
Comment (michalvavrik) [reply docs/guides/operator/advanced-configuration.adoc]: done

File: docs/guides/operator/advanced-configuration.adoc (hunk 1 above)
Comment (ahus1) [reply docs/guides/operator/advanced-configuration.adoc]: Pointing to other docs in the Keycloak docs and a list of configuration properties is ok. 

Pointing to (upstream) docs like Quarkus / Agroal docs is IMHO not ok as it doesn't work for downstream. And we've seen that people try a lot of options from those other docs, that might have unforseen complications on Keycloak, which we then need to explain later to them that we don't support them, or we break their setup once we change Keycloak.

Comment (michalvavrik) [reply docs/guides/operator/advanced-configuration.adoc]: > Pointing to other docs in the Keycloak docs and a list of configuration properties is ok.
> 
> Pointing to (upstream) docs like Quarkus / Agroal docs is IMHO not ok as it doesn't work for downstream. And we've seen that people try a lot of options from those other docs, that might have unforseen complications on Keycloak, which we then need to explain later to them that we don't support them, or we break their setup once we change Keycloak.

Ok, thanks for the info. IMO it is fine as is now, but if you have more feedback or suggestions, just go let me know and I'll look into it.

**Review by ahus1 — APPROVED** [review]
Thank you for these changes!

**Review by mabartos — APPROVED** [review]
@michalvavrik LGTM, thanks!
